CommonConcepts/Plugins/Rhetos.Dom.DefaultConcepts/Persistence/FullTextSearchWithRankInterceptor.cs

```diff
@@ -0,0 +1,142 @@
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Data.Entity.Infrastructure.Interception;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Data.Common;
+using System.Data;
+using System.Text.RegularExpressions;
+
+namespace Rhetos.Dom.DefaultConcepts.Persistence
+{
+    /// <summary>
+    /// Based on http://www.entityframework.info/Home/FullTextSearch.
+    /// This interceptor modifies SQL query generated by FullTextSearchId function mapping in DatabaseExtensionFunctionsMapping.
+    /// </summary>
+    public class FullTextSearchWithRankInterceptor : IDbCommandInterceptor
+    {
+        #region IDbCommandInterceptor implementation
+
+        public void NonQueryExecuted(DbCommand command, DbCommandInterceptionContext<int> interceptionContext)
+        {
+        }
+
+        public void NonQueryExecuting(DbCommand command, DbCommandInterceptionContext<int> interceptionContext)
+        {
+        }
+
+        public void ReaderExecuted(DbCommand command, DbCommandInterceptionContext<DbDataReader> interceptionContext)
+        {
+        }
+
+        public void ReaderExecuting(DbCommand command, DbCommandInterceptionContext<DbDataReader> interceptionContext)
+        {
+            RewriteFullTextQuery(command);
+        }
+
+        public void ScalarExecuted(DbCommand command, DbCommandInterceptionContext<object> interceptionContext)
+        {
+        }
+
+        public void ScalarExecuting(DbCommand command, DbCommandInterceptionContext<object> interceptionContext)
+        {
+            RewriteFullTextQuery(command);
+        }
+
+        #endregion
+
+        private static void RewriteFullTextQuery(DbCommand cmd)
+        {
+            if (cmd.CommandText.Contains(DatabaseExtensionFunctions.InterceptFullTextSearchWithRankFunction))
+            {
+                var parseFtsQuery = new Regex(@"\(\[Rhetos\]\.\[" + DatabaseExtensionFunctions.InterceptFullTextSearchWithRankFunction + @"\]\((?<itemId>.+?), (?<pattern>.*?), (?<rankTop>.*?), (?<tableName>.*?), (?<searchColumns>.*?)\)\) = 1", RegexOptions.Singleline);
+
+                var ftsQueries = parseFtsQuery.Matches(cmd.CommandText).Cast<Match>();
+
+                // Note: When modifying regex, update the error description below!
+                var itemIdFormat = new Regex(@"^(\[\w+\]|\w+)(\.(\[\w+\]|\w+))+$", RegexOptions.Singleline);
+                var tableNameFormat = new Regex(@"^N'(?<unquoted>(\[\w+\]|\w+)(\.(\[\w+\]|\w+))+)'$", RegexOptions.Singleline);
+                var searchColumnsFormat = new Regex(@"^N'(?<unquoted>(\w+|\(\w+(\s*,\s*\w+)*\)|\*))'$", RegexOptions.Singleline);
+                var patternFormat = new Regex(@"^(\@\w+|N'([^']*('')*)*')$", RegexOptions.Singleline);
+                var rankFormat = new Regex(@"^(\@\w+|N'([^']*('')*)*')$", RegexOptions.Singleline);

```

Comment:
Please add a unit test with `rankTop` parameter provided is an integer literal instead of a variable. 
For example: `DatabaseExtensionFunctions.FullTextSearchWithRank(item.ID, test.Key, 2, "TestFullTextSearch.Simple_Search", "*"))`

There might be a problem with `rankFormat` regex when the `rankTop` parameter is a literal. I am guessing that it should be modified from `^(\@\w+|N'([^']*('')*)*')$` to `^(\@\w+|\d+)$`, or something similar